packages/access-client/src/agent-use-cases.js

```diff
@@ -92,20 +92,37 @@ export async function addProvider({ access, space, account, provider }) {
 }
 
 /**
+ * @typedef {(delegations: Ucanto.Delegation<Ucanto.Capabilities>[]) => boolean} DelegationsChecker
+ */
+
+/**
+ * @type DelegationsChecker
+ */
+export function delegationsIncludeSessionProof(delegations) {
+  return delegations.some((d) => isSessionProof(d))
+}
+
+/**
+ * @param {DelegationsChecker} delegationsMatch
  * @param {AccessAgent} access
  * @param {Ucanto.DID} delegee
  * @param {object} [options]
  * @param {number} [options.interval]
  * @param {AbortSignal} [options.abort]
  * @returns {Promise<Iterable<Ucanto.Delegation>>}
  */
-export async function expectNewClaimableDelegations(access, delegee, options) {
```

Comment:
maybe we should leave `expectNewClaimableDelegations` in to maintain back compat?
idc either way. i.e. if we remove it and need it again, we can add it.
but if we remove it, might want to note that the export will go away (everything in this file is exported, which we might wanna change here by dropping the `export * from "./agent-use-cases.js"` in ./agent.js)

Reply:
also fine to me to remove `expectNewClaimableDelegations` here. we can add it back in if needed